app/models/driver.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class Driver < ApplicationRecord
+  has_many :trips
+  validates :name, presence: true
+  validates :vin, presence: true
+
+  def average_rating
+    sum_of_ratings = 0.00
+    average = 0.00
+    if self.trips.empty?
+      average
+    else
+      self.trips.map do |trip|
+        if trip.rating.nil?
+          next
+        else
+          sum_of_ratings += trip.rating.to_f
+        end
+      end
```

Comment:
Since you're just doing a total not a conversion of the list, either do it with `.each` instead of `.map` or do this

```suggestion
      sum_of_ratings = self.trips.map { |trip|
        if trip.rating.nil?
          0.0
        else
          trip.rating.to_f
        end
      }.sum
```